projects/test_project/rules.mk

```diff
@@ -1,8 +1,6 @@
-# Specify the device family to build for
-DEVICE_FAMILY := stm32f0xx
-
 # Specify the main file (relative to this file if not in this folder)
 MAIN := main.c
 
 # Specify the device library you want to include
-DEVICE_LIBRARY := libstm32f0.a
+DEPS := libraries/stm32f0xx/stm32f0 \
```

Comment:
I think we should enforce libraries as things that should be project-independent. One of things I disliked about all the submodules were the variants of ms-lib.

Multiple configurations should be handled on a per-project basis. If necessary, people can modify the project's makefile to change which config they're building with.